### PR TITLE
Add Appveyor build

### LINKED ISSUE #161: Add Appveyor as CI for project
We have some issues on windows, like #129, to avoid this - need to add windows testing.

### PR DESCRIPTION
**Pull request by menshikh-iv**
Just add windows builds by @mpenkov request

- Fix #161

### CODE REVIEW

**Comment by menshikh-iv** [comment]
@mpenkov @piskvorky I need an access from appveyor to the current repo, can you make this?

**Comment by mpenkov** [comment]
OK, added the webhook, please check.

Also, please add a badge to the README. https://ci.appveyor.com/project/piskvorky/smart-open/settings/badges

**Comment by menshikh-iv** [comment]
@mpenkov, unfortunately, doesn't work :(

**Comment by piskvorky** [comment]
-1 on the badge. Why do we need that? We have too many badges already, and users couldn't care less about AppVeyor.

See also https://github.com/RaRe-Technologies/gensim/pull/2772.

**Comment by mpenkov** [comment]
I had the discussion with @menshikh-iv re the counter. It includes:

- everyone who uses smart_open directly (our direct user base)
- everyone who uses gensim (which requires smart_open as a dependency)
- everyone who uses something that uses gensim

Also, keep in mind that these aren't necessarily human users. e.g. if a script initializes a virtual environment and pip-installs smart_open into it, that's a download.

**Comment by mpenkov** [comment]
re. the badge, I reckon we _do_ need it, because we want to know our builds are working, both for Linux _and_ Windows. Yeah, ideally, we'll see things break _before_ we merge PRs, but there is still a chance that a breaking change will make it into develop (or master *gulp*). Having that badge there will prevent us making releases with a breaking change (in fact, it's one of the things on the gensim release checklist: check the badges).

File: README.rst [hunk 1]
```diff
@@ -2,14 +2,16 @@
 smart_open — utils for streaming large files in Python
 ======================================================
 
-|License|_ |Travis|_ |Coveralls|_ |Downloads|_
+|License|_ |Travis|_ |Appveyor|_ |Coveralls|_ |Downloads|_
```
Comment (mpenkov) [inline README.rst:5]:
```suggestion

|License|_ |Travis|_ |Appveyor|_ |Coveralls|_ |Downloads|_
```

**Comment by piskvorky** [comment]
Of course these downloads are bots. But even then, the numbers are weird. The count used to be a few 10Ks, then jumped to 2M a few weeks ago, and now 5M… Something's going on.

> we do need it, because we want to know our builds are working, both for Linux and Windows.

Hm, I'd prefer that handled as part of our PR / merge process. It's an internal developer thing, not the most important thing to users = very top of README.

Not super critical, just more irrelevant noise.

**Comment by mpenkov** [comment]
@piskvorky We need you to enable smart_open in AppVeyor using the credentials for the RaRe-Technologies organization.

> Go to "Account -> Authorizations", then "Revoke access" and finally do "Import existing installations".

Can you please give that a try and let me know what happens?

**Comment by piskvorky** [comment]
Done.

Screen Shot 2020-04-22 at 10 00 44

**Comment by mpenkov** [comment]
Thanks @piskvorky , that did the trick.

@menshikh-iv Appveyor picks up our commits now. There's a problem with the yaml file you added, please have a look: https://ci.appveyor.com/project/piskvorky/smart-open

**Comment by mpenkov** [comment]
Fix tests that are broken on Appveyor.

They are broken for various reasons. It's likely that the problem is with the tests themselves (e.g. the NamedTemporaryFile problem fixed in this PR) as opposed to smart_open itself, but it's worth investigating in more detail (at some later point).

**Comment by mpenkov** [comment]
Write a Windows .BAT equivalent of travis_ci_helpers.sh

We need this .BAT file for running integration tests under Appveyor, which is unable to run our existing bash script.

Alternatively, rewrite that script in Python so we can use it under both Travis and Appveyor.

**Comment by mpenkov** [comment]
Merging. Good work @menshikh-iv !!!

**Pull request merged**